src/crimson/osd/recovery_backend.cc

```diff
@@ -238,18 +238,17 @@ RecoveryBackend::scan_for_backfill(
 	if (pg.is_primary()) {
 	  obc = pg.obc_registry.maybe_get_cached_obc(object);
 	}
-	if (obc) {
-	  if (obc->obs.exists) {
-	    logger().debug("scan_for_backfill found (primary): {}  {}",
-			   object, obc->obs.oi.version);
-	    version_map->emplace(object, obc->obs.oi.version);
-	  } else {
-	    // if the object does not exist here, it must have been removed
-	    // between the collection_list_partial and here.  This can happen
-	    // for the first item in the range, which is usually last_backfill.
-	  }
```

Comment:
Can you please explain why this case is no longer relevant?

Reply:
I think this case still exists, in which the later "load_metadata" also returns an empty obc and the object won't be put into the backfill interval. So it looks to me that this PR doesn't conflict with the main branch in this case, am I right?